commands/server/start.go

```diff
@@ -74,11 +75,18 @@ func StartCmd(gen AppGenerator, logger log.Logger, home string, args []string) e
 	}
 	svr.SetLogger(logger.With("module", "abci-server"))
 	svr.Start()
-
-	// Wait forever
-	cmn.TrapSignal(func() {
+	ctx, cancel := context.WithCancel(context.Background())
+	cmn.TrapSignal(logger, func() {
```

Comment:
Ah, TrapSignal no longer blocks. 
Nice changes here, but I see no need to use Context and a simpler solution would be more clear (that this is not a context we pass into the various rpc functions).

```golang
done := make(channel bool, 1)
cmn.TrapSignal(logger, func() {
  svr.Stop()
  done <- true
})
// wait forever
<- done
```

Unless you can explain the advantage of the current setup (maybe there is a subtle point I miss... not sure why the select in for loop... seems like a busy wait to me)

Reply:
yeah, I'll do this. Easier indeed